common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -127,4 +127,15 @@ private void appendUnit(StringBuilder sb, long value, String unit) {
    * @throws ArithmeticException if a numeric overflow occurs
    */
   public Duration extractAsDuration() { return Duration.of(microseconds, ChronoUnit.MICROS); }
+
+  @Override
+  public int compareTo(CalendarInterval o) {
+    if (this.months != o.months) {
```

Comment:
Comparing intervals does not necessarily short circuits via months. We could result in `1 month > 0 months 32 days`, which is wrong, obviously.

Besides, 1 month can be 28 ~ 30 days, making the legacy calendar interval type uncomparable

Reply:
We should add some comments to explain that this is alphabet ordering. It does not have actual meaning but just makes it possible to find identical interval instances.

Reply:
@stefankandic did you generate this using IDEA?

Reply:
Added the comments.

@cloud-fan method was generated by intellij but I implemented the logic